Simplified/NYPLUserAccount.swift

```diff
@@ -154,222 +312,89 @@ extension Notification.Name {
-      return
-    }
-    
-    sharedKeychain.setObject(token, forKey: adobeTokenKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    _adobeToken.safeWrite(token)
+    notifyAccountDidChange()
   }
   
   @objc(setLicensor:)
   func setLicensor(_ licensor: [String : Any]) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(licensor, forKey: licensorKey)
+    _licensor.safeWrite(licensor)
   }
   
   @objc(setAuthorizationIdentifier:)
   func setAuthorizationIdentifier(_ identifier: String) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(identifier, forKey: authorizationIdentifierKey)
+    _authorizationIdentifier.safeWrite(identifier)
   }
   
   @objc(setPatron:)
   func setPatron(_ patron: [String : Any]) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(patron, forKey: patronKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    _patron.safeWrite(patron)
+    notifyAccountDidChange()
   }
   
   @objc(setAuthToken:)
   func setAuthToken(_ token: String) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(token, forKey: authTokenKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    credentials = .token(authToken: token)
   }
   
   @objc(setProvider:)
   func setProvider(_ provider: String) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(provider, forKey: providerKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    _provider.safeWrite(provider)
+    notifyAccountDidChange()
   }
   
   @objc(setUserID:)
   func setUserID(_ id: String) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(id, forKey: userIDKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    _userID.safeWrite(id)
+    notifyAccountDidChange()
   }
   
   @objc(setDeviceID:)
   func setDeviceID(_ id: String) {
-    guard let sharedKeychain = NYPLKeychain.shared() else {
-      return
-    }
-    
-    sharedKeychain.setObject(id, forKey: deviceIDKey)
-    
-    NotificationCenter.default.post(
-      name: Notification.Name.NYPLUserAccountDidChange,
-      object: self
-    )
+    _deviceID.safeWrite(id)
+    notifyAccountDidChange()
   }
     
   // MARK: - Remove
-    
-  func removeBarcodeAndPIN() {
```

Comment:
I believe Kyle said we should keep this because it's used in OpenEbooks, and we do want to marge the apps at some point.

Reply:
Open Ebooks codebase is very old, based on the current logic I'd rather suggest to copy-paste some of their UI classes (ex. the one presenting pop up with buttons to login) to current SimplyE app instead of merging.

I didn't see this method used anywhere else in SimplyE so I just merged them, I can split them back

Reply:
yeah for now I would split them back... I had removed it too :) easy to miss, currently before removing I check the OpenEbooks code base to verify it's not used there. I do realize this is making us move slowly and it's not YAGNI... come to think of it.... maybe we _can_ remove it since the equivalent OpenEbooks code is still in Objc, so there will be conflicts anyway.